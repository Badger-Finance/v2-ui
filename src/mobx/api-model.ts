/**
 * All API Model Objects should be directly in line with API interface model
 * objects to ensure data compatiblity. Interfaces located at link below:
 * https://github.com/axejintao/badger-api/tree/staging/src/interface
 */
import BigNumber from "bignumber.js";

export interface Geyser {
  emissions: Emission[],
}

export interface Token {
  address: string;
  name: string;
  symbol: string;
  decimals: number;
}

export interface Emission {
  token: Token,
  unlockSchedule: UnlockSchedule,
}

export interface UnlockSchedule {
	initialLocked: BigNumber,
	endAtSec: BigNumber,
	durationSec: BigNumber,
	startTime: BigNumber,
}

export interface TokenBalance {
  address: string;
  name: string;
  symbol: string;
  decimals: number;
  balance: number,
  value: number,
}

export interface ValueSource {
  name: string,
  apy: number,
  performance: Performance,
}

export interface Sett {
  name: string,
  asset: string,
  ppfs: number,
  value: number,
  apy: number,
  tokens: TokenBalance[],
  sources: ValueSource[],
  geyser?: Geyser,
}

export interface SettSummary {
  name: string,
  asset: string,
  value: number,
  tokens: TokenBalance[],
}
