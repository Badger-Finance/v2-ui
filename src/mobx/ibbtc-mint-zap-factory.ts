import { IbBTCMintZap } from './model/vaults/ibbtc-mint-zap';
import { RootStore } from './RootStore';
import addresses from 'config/ibBTC/addresses.json';
import { TokenZap } from './model/vaults/token-zap';
import { GeneralVaultZap } from './model/vaults/general-vault-zap';
import { RenVaultZap } from './model/vaults/ren-vault-zap';
import { Token } from '@badger-dao/sdk';

export class IbBTCMintZapFactory {
  private static tokenZaps: Record<string, TokenZap> = {};
  private static generalVaultZaps: Record<string, GeneralVaultZap> = {};
  private static renVaultZaps: Record<string, RenVaultZap> = {};

  static getIbBTCZap(store: RootStore, token: Token): IbBTCMintZap {
    if (addresses.mainnet.contracts.TokenZap.supportedTokens.includes(token.address)) {
      if (!this.tokenZaps[token.address]) {
        this.tokenZaps[token.address] = new TokenZap(store, token);
      }
      return this.tokenZaps[token.address];
    }

    if (addresses.mainnet.contracts.GeneralVaultZap.supportedTokens.includes(token.address)) {
      if (!this.generalVaultZaps[token.address]) {
        this.generalVaultZaps[token.address] = new GeneralVaultZap(store, token);
      }
      return this.generalVaultZaps[token.address];
    }

    if (addresses.mainnet.contracts.RenVaultZap.supportedTokens.includes(token.address)) {
      if (!this.renVaultZaps[token.address]) {
        this.renVaultZaps[token.address] = new RenVaultZap(store, token);
      }
      return this.renVaultZaps[token.address];
    }

    throw new Error(`Unrecognized peak type for token => ${token.symbol}`);
  }
}
