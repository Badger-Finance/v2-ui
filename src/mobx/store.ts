import { RouterStore } from 'mobx-router';
import UiState from './reducers';
import WalletStore from './stores/walletStore';
import ContractsStore from './stores/contractsStore';
import AirdropStore from './stores/airdropStore';
import RebaseStore from './stores/rebaseStore';
import RewardsStore from './stores/rewardsStore';
import IbBTCStore from './stores/ibBTCStore';
import TransactionsStore from './stores/transactionsStore';
import SettStoreV2 from './stores/settStoreV2';
import { NETWORK_LIST } from '../config/constants';
import HoneyPotStore from './stores/honeyPotStore';

export class RootStore {
	public router: RouterStore<RootStore>;
	public wallet: WalletStore;
	public uiState: UiState;
	public contracts: ContractsStore;
	public airdrops: AirdropStore;
	public rebase: RebaseStore;
	public rewards: RewardsStore;
	public ibBTCStore: IbBTCStore;
	public setts: SettStoreV2;
	public transactions: TransactionsStore;
	public honeyPot: HoneyPotStore;

	constructor() {
		this.router = new RouterStore<RootStore>(this);
		this.wallet = new WalletStore(this);
		this.contracts = new ContractsStore(this);
		this.airdrops = new AirdropStore(this);
		this.rebase = new RebaseStore(this);
		this.rewards = new RewardsStore(this);
		this.uiState = new UiState(this);
		this.ibBTCStore = new IbBTCStore(this);
		this.honeyPot = new HoneyPotStore(this);
		// RenVM transactions store.
		this.transactions = new TransactionsStore(this);
		this.setts = new SettStoreV2(this);

		this.walletRefresh();
	}

	async walletRefresh(): Promise<void> {
		const chain = this.wallet.network.name;
		const refreshData = [
			this.setts.loadAssets(chain),
			this.setts.loadPrices(chain),
			this.wallet.getGasPrice(),
			this.contracts.updateProvider(),
		];
		if (chain === NETWORK_LIST.ETH) {
			refreshData.push(this.setts.loadGeysers(chain));
			refreshData.push(this.rebase.fetchRebaseStats());
		} else {
			refreshData.push(this.setts.loadSetts(chain));
		}
		await Promise.all(refreshData);

		if (this.wallet.connectedAddress) {
			this.contracts.updateProvider();
			await this.wallet.getGasPrice();
			await this.contracts.fetchContracts();
			if (chain === NETWORK_LIST.ETH) {
				this.uiState.reduceRebase();
				this.ibBTCStore.init();
				this.rewards.fetchSettRewards();
				this.uiState.reduceTreeRewards();
				this.airdrops.fetchAirdrops();
				this.uiState.reduceAirdrops();
			}
			this.uiState.reduceStats();
		}
	}
}

const store = new RootStore();

export default store;
