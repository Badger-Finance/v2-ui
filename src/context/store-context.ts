import { createContext } from "react";
import { RootStore } from "../mobx/store";

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;