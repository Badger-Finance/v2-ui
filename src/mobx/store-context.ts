import { RootStore } from './store';
import { createContext } from 'react';

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;
