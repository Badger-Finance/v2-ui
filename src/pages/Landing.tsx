import CurrencyInfoCard from '../components-v2/common/CurrencyInfoCard';
import CurrencyPicker from '../components-v2/landing/CurrencyPicker';
import SamplePicker from '../components-v2/landing/SamplePicker';
import WalletSlider from '../components-v2/landing/WalletSlider';
import { Grid, makeStyles, Button } from '@material-ui/core';
import PageHeader from '../components-v2/common/PageHeader';
import { StoreContext } from '../mobx/store-context';
import { observer } from 'mobx-react-lite';
import React, { useContext } from 'react';
import BigNumber from 'bignumber.js';
import SettList from 'components-v2/landing/SettList';
import { RewardsModal } from '../components-v2/landing/RewardsModal';
import { SettState } from '../mobx/model/setts/sett-state';
import { HeaderContainer, LayoutContainer } from '../components-v2/common/Containers';
import { FLAGS } from 'config/constants';

const useStyles = makeStyles((theme) => ({
	marginTop: {
		marginTop: theme.spacing(3),
	},
	rewardContainer: {
		marginTop: theme.spacing(3),
		marginBottom: theme.spacing(3),
		marginLeft: 'auto',
		marginRight: 'auto',
		display: 'flex',
		flexDirection: 'column',
	},
	widgetContainer: {
		display: 'flex',
		justifyContent: 'space-between',
		height: '2.2rem',
		marginBottom: theme.spacing(1),
		[theme.breakpoints.down('xs')]: {
			flexDirection: 'column',
			alignItems: 'center',
			justifyContent: 'center',
			marginBottom: theme.spacing(3),
		},
	},
	rewardText: {
		marginRight: theme.spacing(3),
		textAlign: 'left',
	},
	statPaper: {
		padding: theme.spacing(2),
	},
	rewardItem: {
		padding: 0,
		flexWrap: 'wrap',
	},
	walletContainer: {
		[theme.breakpoints.down('xs')]: {
			marginTop: theme.spacing(1),
		},
	},
	pickerContainer: {
		display: 'flex',
		marginRight: theme.spacing(1),
		alignItems: 'flex-end',
		[theme.breakpoints.down('xs')]: {
			marginBottom: theme.spacing(2),
			marginTop: theme.spacing(-1),
		},
	},
	announcementButton: {
		marginTop: theme.spacing(3),
		pointerEvents: 'none',
	},
	linkButton: {
		marginTop: theme.spacing(3),
	},
}));

interface LandingProps {
	title: string;
	subtitle?: string;
	state: SettState;
}

const Landing = observer((props: LandingProps) => {
	const classes = useStyles();
	const store = useContext(StoreContext);
	const { title, subtitle, state } = props;

	const {
		wallet: { connectedAddress },
		uiState: { currency },
		network: { network },
		setts,
		prices,
		user,
	} = store;
	const { protocolSummary } = setts;
	const userConnected = !!connectedAddress;

	const badgerToken = network.deploy.token.length > 0 ? network.deploy.token : undefined;
	const totalValueLocked = protocolSummary ? new BigNumber(protocolSummary.totalValue) : undefined;
	const badgerPrice = badgerToken ? prices.getPrice(badgerToken) : undefined;
	const portfolioValue = userConnected ? user.portfolioValue : undefined;

	return (
		<LayoutContainer>
			{/* Landing Metrics Cards */}
			<Grid container spacing={1} justify="center">
				<HeaderContainer item xs={12}>
					<PageHeader title={title} subtitle={subtitle} />
				</HeaderContainer>
				<Grid item xs={12} className={classes.widgetContainer}>
					<div className={classes.walletContainer}>{userConnected && <WalletSlider />}</div>
					<div className={classes.pickerContainer}>
						<RewardsModal />
						<SamplePicker />
						<CurrencyPicker />
					</div>
				</Grid>
				<Grid item xs={12} md={userConnected ? 4 : 6}>
					<CurrencyInfoCard title="Total Value Locked" value={totalValueLocked} currency={currency} />
				</Grid>
				{userConnected && (
					<Grid item xs={12} md={4}>
						<CurrencyInfoCard title="Your Portfolio" value={portfolioValue} currency={currency} />
					</Grid>
				)}
				<Grid item xs={12} md={userConnected ? 4 : 6}>
					<CurrencyInfoCard title="Badger Price" value={badgerPrice} currency={currency} />
				</Grid>
			</Grid>

			{state === SettState.Guarded && (
				<Grid container spacing={1} justify="center">
					<Button className={classes.announcementButton} size="small" variant="outlined" color="primary">
						Note: New Vaults may take up to 2 weeks from launch to reach full efficiency.
					</Button>
				</Grid>
			)}
			{state === SettState.Open && FLAGS.BOOST_OPTIMIZER && (
				<Grid container spacing={1} justify="center">
					<Button
						className={classes.linkButton}
						size="small"
						variant="contained"
						color="primary"
						onClick={() =>
							window.open(
								'https://medium.com/badgerdao/badger-boost-power-up-stake-ratio-levels-e0c9802fc5c3',
							)
						}
					>
						Badger Boost Power Up has been implemented - Click here to learn more
					</Button>
				</Grid>
			)}

			<SettList state={state} />
		</LayoutContainer>
	);
});

export default Landing;
