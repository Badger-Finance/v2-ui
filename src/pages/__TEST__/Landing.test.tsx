import React from 'react';
import { render, fireEvent, cleanup, screen, act } from '@testing-library/react';
import Landing from '../Landing';
import '@testing-library/jest-dom';
import { StoreProvider } from '../../mobx/store-context';
import store from '../../mobx/store';

afterEach(cleanup);

describe('Landing Page', () => {
	const connectedStore = store;
	act(() => {
		connectedStore.wallet.connectedAddress = '0x1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a';
	});

	test('Renders correctly', () => {
		const { container } = render(
			<StoreProvider value={connectedStore}>
				<Landing />
			</StoreProvider>,
		);
		expect(container).toMatchSnapshot();
	});

	test('Clicking portfolio switch shows empty portfolio', async () => {
		const { container } = render(
			<StoreProvider value={connectedStore}>
				<Landing />
			</StoreProvider>,
		);
		// Clicks on switch
		await act(async () => {
			await fireEvent.click(screen.getByText('Portfolio View'));
		});
		expect(container).toMatchSnapshot();
	});

	// TODO: Create a mock Web3 provider to emulate connection and test further interactions
	/* 
	test('Selecting different currency changes displayed currency', async () => {
		store.wallet.connectedAddress = '';
		const { getByRole, findByText } = render(
			<StoreProvider value={store}>
				<Landing />
			</StoreProvider>,
		);
		// Opens currency menu
		await fireEvent.mouseDown(getByRole('button', { name: defaultCurrency }));

		// Checks that CAD is in menu
		const currency = await findByText('CAD');
		expect(currency).toBeVisible;

		// Selects CAD
		await fireEvent.click(currency);

		// Finds ocurrance of C$
		const ocurrance = await findByText(/C\$/i);
		expect(ocurrance).toBeVisible;
	}); */
});
