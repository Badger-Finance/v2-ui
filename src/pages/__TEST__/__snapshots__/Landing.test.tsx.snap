// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Clicking portfolio switch shows empty portfolio 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-landingContainer-38 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root makeStyles-headerContainer-39 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="makeStyles-headerContainer-48"
        >
          <div
            class="makeStyles-headerContent-49"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary"
            >
              Sett Vaults
            </h2>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
            >
              Powerful Bitcoin strategies. Automatic staking rewards.
            </h6>
          </div>
          <div
            class="makeStyles-headerWidgets-47"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary makeStyles-gasSelector-50 MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="cursor: pointer; font-size: 1.2rem; margin-right: .8rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.77 7.23l.01-.01-3.72-3.72L15 4.56l2.11 2.11c-.94.36-1.61 1.26-1.61 2.33 0 1.38 1.12 2.5 2.5 2.5.36 0 .69-.08 1-.21v7.21c0 .55-.45 1-1 1s-1-.45-1-1V14c0-1.1-.9-2-2-2h-1V5c0-1.1-.9-2-2-2H6c-1.1 0-2 .9-2 2v16h10v-7.5h1.5v5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V9c0-.69-.28-1.32-.73-1.77zM12 10H6V5h6v5zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                />
              </svg>
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
                role="button"
                tabindex="0"
              >
                75
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="standard"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-51 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-52"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-walletButton-58 MuiButton-containedSecondary MuiButton-disableElevation"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                0x1a1a1...a1a1a1a
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <div
                    class="makeStyles-walletDot-55 makeStyles-greenDot-57"
                  />
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-widgetContainer-42 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div>
          <label
            class="MuiFormControlLabel-root makeStyles-walletSlider-59"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-60 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-61 Mui-checked"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-63 MuiSwitch-input"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Portfolio View
            </span>
          </label>
        </div>
        <div
          class="makeStyles-pickerContainer-46"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-samplePicker-64"
            style="margin-top: auto; margin-bottom: auto;"
            title="ROI combines the appreciation of the vault with its $BADGER or $DIGG emissions. All numbers are an approximation based on historical data."
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              YEAR
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="year"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-51 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-52"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-currencyPicker-65"
            style="margin-top: auto; margin-bottom: auto;"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              USD
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="usd"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-51 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-52"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root makeStyles-infoPaper-66 MuiPaper-elevation2 MuiPaper-rounded"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            Total Value Locked
          </h6>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-wave"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Placeholder
            </h5>
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root makeStyles-infoPaper-66 MuiPaper-elevation2 MuiPaper-rounded"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            Your Portfolio
          </h6>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-wave"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Placeholder
            </h5>
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root makeStyles-infoPaper-66 MuiPaper-elevation2 MuiPaper-rounded"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            Badger Price
          </h6>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-wave"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Placeholder
            </h5>
          </span>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-settListContainer-74"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        style="margin: 2rem 0px;"
      >
        Your address does not have tokens to deposit.
      </h6>
      <div />
    </div>
  </div>
</div>
`;

exports[`Landing Page Renders correctly 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-landingContainer-1 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root makeStyles-headerContainer-2 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="makeStyles-headerContainer-11"
        >
          <div
            class="makeStyles-headerContent-12"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary"
            >
              Sett Vaults
            </h2>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
            >
              Powerful Bitcoin strategies. Automatic staking rewards.
            </h6>
          </div>
          <div
            class="makeStyles-headerWidgets-10"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary makeStyles-gasSelector-13 MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="cursor: pointer; font-size: 1.2rem; margin-right: .8rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.77 7.23l.01-.01-3.72-3.72L15 4.56l2.11 2.11c-.94.36-1.61 1.26-1.61 2.33 0 1.38 1.12 2.5 2.5 2.5.36 0 .69-.08 1-.21v7.21c0 .55-.45 1-1 1s-1-.45-1-1V14c0-1.1-.9-2-2-2h-1V5c0-1.1-.9-2-2-2H6c-1.1 0-2 .9-2 2v16h10v-7.5h1.5v5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V9c0-.69-.28-1.32-.73-1.77zM12 10H6V5h6v5zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                />
              </svg>
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
                role="button"
                tabindex="0"
              >
                75
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="standard"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-15"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-walletButton-21 MuiButton-containedSecondary MuiButton-disableElevation"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                0x1a1a1...a1a1a1a
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <div
                    class="makeStyles-walletDot-18 makeStyles-greenDot-20"
                  />
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-widgetContainer-5 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div>
          <label
            class="MuiFormControlLabel-root makeStyles-walletSlider-22"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-23 MuiSwitch-switchBase MuiSwitch-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-26 MuiSwitch-input"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Portfolio View
            </span>
          </label>
        </div>
        <div
          class="makeStyles-pickerContainer-9"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-samplePicker-27"
            style="margin-top: auto; margin-bottom: auto;"
            title="ROI combines the appreciation of the vault with its $BADGER or $DIGG emissions. All numbers are an approximation based on historical data."
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              YEAR
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="year"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-15"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-currencyPicker-28"
            style="margin-top: auto; margin-bottom: auto;"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              USD
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="usd"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-15"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root makeStyles-infoPaper-29 MuiPaper-elevation2 MuiPaper-rounded"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            Total Value Locked
          </h6>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-wave"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Placeholder
            </h5>
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root makeStyles-infoPaper-29 MuiPaper-elevation2 MuiPaper-rounded"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            Your Portfolio
          </h6>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-wave"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Placeholder
            </h5>
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root makeStyles-infoPaper-29 MuiPaper-elevation2 MuiPaper-rounded"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            Badger Price
          </h6>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-wave"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              Placeholder
            </h5>
          </span>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-settListContainer-37"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        style="margin: 2rem 0px;"
      >
        Loading Badger Setts...
      </h6>
      <div />
    </div>
  </div>
</div>
`;
